internal/engine/docker.go

```diff
@@ -45,12 +46,29 @@ func dockerImageProvider(ctx context.Context, runnerHost *url.URL) (string, erro
 		// We only have a tag in the image ref, so resolve it to a digest. The default
 		// auth keychain parses the same docker credentials as used by the buildkit
 		// session attachable.
-		img, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+		if img, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain)); err != nil {
+			fmt.Fprintf(os.Stderr, "failed to resolve image digest: %s, falling back to leftover engine\n", err)
+			fallbackToLeftoverEngine = true
+		} else {
+			id = img.Digest.String()
+		}
+	}
+
+	// We collect leftover engine anyway since we garbage collect them at the end
+	// And check if we are in a fallback case then perform fallback to most recent engine
+	leftoverEngines, err := collectLeftoverEngines(ctx)
+	if fallbackToLeftoverEngine {
 		if err != nil {
-			return "", errors.Wrap(err, "resolving image digest")
+			return "", errors.Wrap(err, "failed to list containers")
+		}
```

Comment:
I don't think this is ever expected to fail, even if there are no matching containers it should still succeed. So I think we should move the `if err != nil` check right above the `if fallbackToLeftoverEngine` check

Reply:
I did this change. I just an error message so it can still try to start the container if we are in the normal case